libbeat/common/kubernetes/metadata/pod.go

```diff
@@ -89,3 +91,14 @@ func (p *pod) GenerateFromName(name string, opts ...FieldOptions) common.MapStr
 
 	return nil
 }
+
+func (p *pod) fixLabels(in common.MapStr) common.MapStr {
```

Comment:
Hmm.. "fix" says nothing about the behavior. Could you rename this method to sth less vague? `movePodLabels`, `extractPodLabel`, etc.